src/components/backgroundselector/BackgroundService.js

```diff
@@ -114,9 +116,49 @@ goog.require('ga_urlutils_service');
         } */
       };
 
+      function updateBgLayerStyleUrlParam(bgLayer) {
+        var styleUrlValue = bgLayer.externalStyleUrl;
+        if (styleUrlValue) {
+          // Save the url in the bg config to get it
+          // when we switch back.
+          getBgById(bgLayer.id).styleUrl = styleUrlValue;
+          gaPermalink.updateParams({
+            bgLayer_styleUrl: styleUrlValue
+          });
+        } else {
+          gaPermalink.deleteParam('bgLayer_styleUrl');
+        }
+      }
+
+      // Update permalink on bgLayer's modification
+      var registerBgLayerStyleUrlPermalink = function(scope, map) {
+        var deregFns = [];
+        scope.layers = map.getLayers().getArray();
+        scope.layerFilter = gaLayerFilters.background;
+        scope.$watchCollection('layers | filter:layerFilter',
+            function(layers) {
+
+              // deregister the listeners we have on each layer and register
+              // new ones for the new set of layers.
+              angular.forEach(deregFns, function(deregFn) {
+                deregFn();
+              });
+              deregFns.length = 0;
+
+              angular.forEach(layers, function(layer) {
+                deregFns.push(scope.$watch(function() {
+                  return layer.externalStyleUrl;
+                }, function() {
+                  updateBgLayerStyleUrlParam(layer);
+                }));
+              });
+            });
+      }
+
       var Background = function() {
 
         this.init = function(map) {
+          var scope = $rootScope.$new();
```

Comment:
Is it better for perfs to create a new scope? Alternatively we could pass the scope as a parameter.

Reply:
Honestly I don`t know . But since Service is reused all over the app it's good it has its own scope.